pkg/utils/transform/transform.go

```diff
@@ -48,3 +49,54 @@ func InjectNamespaceConditional(preserveNamespace string, targetNamespace string
 		return tf(u)
 	}
 }
+
+func ReplaceKind(fromKind, toKind string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		kind := u.GetKind()
+
+		if kind != fromKind {
+			return nil
+		}
+		err := unstructured.SetNestedField(u.Object, toKind, "kind")
+		if err != nil {
+			return fmt.Errorf("could change resource KIND, %q", err)
+		}
+		return nil
+	}
+}
+
+func InjectLabel(key, value string, overwrite bool, kinds ...string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		kind := u.GetKind()
+		if len(kinds) != 0 && !itemInSlice(kind, kinds) {
```

Comment:
can avoid this check if the `InjectLabel` itself validates that kinds is non-empty.

Reply:
the idea is to allow len(kinds) to be 0 and add lables to any kind

for example, in pipeline-core manifests or addons manifest